### PR TITLE
fix to run clip feature on runOnMainSync

### PR DESCRIPTION
**Pull request by KazuCocoa**
related: https://github.com/appium/ruby_lib_core/pull/69#issuecomment-377466959

When I ran clip feature against Andorid, I encounterd the below error. To avoid the error, I've implemented `runOnMainSync` to run clip feature on main thread.

After this PR, ruby_lib_core's tests run perfectly for clip features on Android.

```
java.lang.RuntimeException: Can't create handler inside thread that has not called Looper.prepare()
  at android.os.Handler.<init>(Handler.java:209)
  at android.os.Handler.<init>(Handler.java:123)
  at android.content.ClipboardManager$2.<init>(ClipboardManager.java:69)
  at android.content.ClipboardManager.<init>(ClipboardManager.java:69)
  at android.app.SystemServiceRegistry$12.createService(SystemServiceRegistry.java:276)
  at android.app.SystemServiceRegistry$12.createService(SystemServiceRegistry.java:275)
  at android.app.SystemServiceRegistry$CachedServiceFetcher.getService(SystemServiceRegistry.java:884)
  at android.app.SystemServiceRegistry.getSystemService(SystemServiceRegistry.java:837)
  at android.app.ContextImpl.getSystemService(ContextImpl.java:1370)
  at io.appium.uiautomator2.utils.ClipboardHelper.getManager(ClipboardHelper.java:37)
  at io.appium.uiautomator2.utils.ClipboardHelper.getTextData(ClipboardHelper.java:45)
  at io.appium.uiautomator2.handler.GetClipboard.safeHandle(GetClipboard.java:61)
  at io.appium.uiautomator2.handler.request.SafeRequestHandler.handle(SafeRequestHandler.java:56)
  at io.appium.uiautomator2.server.AppiumServlet.handleRequest(AppiumServlet.java:238)
  at io.appium.uiautomator2.server.AppiumServlet.handleHttpRequest(AppiumServlet.java:229)
  at io.appium.uiautomator2.http.ServerHandler.channelRead(ServerHandler.java:44)
  at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:366)
  at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:352)
  at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:345)
  at io.netty.handler.codec.MessageToMessageDecoder.channelRead(MessageToMessageDecoder.java:102)
  at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:366)
  at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:352)
  at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:345)
  at io.netty.channel.CombinedChannelDuplexHandler$DelegatingChannelHandlerContext.fireChannelRead(CombinedChannelDuplexHandler.java:435)
  at io.netty.handler.codec.ByteToMessageDecoder.fireChannelRead(ByteToMessageDecoder.java:293)
  at io.netty.handler.codec.ByteToMessageDecoder.channelRead(ByteToMessageDecoder.java:267)
  at io.netty.channel.CombinedChannelDuplexHandler.channelRead(CombinedChannelDuplexHandler.java:250)
  at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:366)
  at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:352)
  at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:345)
  at io.netty.channel.DefaultChannelPipeline$HeadContext.channelRead(DefaultChannelPipeline.java:1294)
  at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:366)
  at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:352)
  at io.netty.channel.DefaultChannelPipeline.fireChannelRead(DefaultChannelPipeline.java:911)
  at io.netty.channel.nio.AbstractNioByteChannel$NioByteUnsafe.read(AbstractNioByteChannel.java:131)
  at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:611)
  at io.netty.channel.nio.NioEventLoop.processSelectedKeysPlain(NioEventLoop.java:514)
  at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:468)
  at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:438)
  at io.netty.util.concurrent.SingleThreadEventExecutor$2.run(SingleThreadEventExecutor.java:140)
  at io.netty.util.concurrent.DefaultThreadFactory$DefaultRunnableDecorator.run(DefaultThreadFactory.jav
```

### CODE REVIEW

**Comment by KazuCocoa** [comment]
@mykola-mokhnach Could you take a look when you have a time?

**Comment by mykola-mokhnach** [comment]
@vmaxim do you have anything to add?

**Comment by mykola-mokhnach** [comment]
Thanks @KazuCocoa

**Comment by KazuCocoa** [comment]
I have no permission to merge this PR, so, if not problem, feel free to merge this 👍

File: app/src/main/java/io/appium/uiautomator2/handler/GetClipboard.java [hunk 1]
```diff
@@ -71,4 +69,30 @@ public AppiumResponse safeHandle(IHttpRequest request) {
...
+        private IHttpRequest request;
+        private AppiumResponse response;
+
+        AppiumGetClipRunnerble(IHttpRequest request) {
+            this.request = request;
+        }
+
+        @Override
+        public void run() {
+            response = new AppiumResponse(getSessionId(request), WDStatus.SUCCESS,
```
Comment (mykola-mokhnach) [inline app/src/main/java/io/appium/uiautomator2/handler/GetClipboard.java]: @KazuCocoa can we here return base64-encoded content and wrap it into AppiumResponse inside handle method? Just for the sake of common code style.

Comment (vmaxim) [reply app/src/main/java/io/appium/uiautomator2/handler/GetClipboard.java]: Also ```switch(contentType)``` can be moved to this ```Runnable```.

File: app/src/main/java/io/appium/uiautomator2/handler/GetClipboard.java [hunk 2]
```diff
@@ -71,4 +69,30 @@ public AppiumResponse safeHandle(IHttpRequest request) {
             return new AppiumResponse(getSessionId(request), WDStatus.UNKNOWN_ERROR, e);
         }
     }
+
+    // Clip feature should run with main thread
+    private AppiumResponse getClipboardTextResponse(final IHttpRequest request) {
+        AppiumGetClipRunnerble runnable = new AppiumGetClipRunnerble(request);
```
Comment (imurchie) [inline app/src/main/java/io/appium/uiautomator2/handler/GetClipboard.java]: I know it is outside of this PR, but can we change `AppiumGetClipRunnerble` to `AppiumGetClipRunnable`?

**Comment by KazuCocoa** [comment]
finished ✅

**Pull request merged**